### PR TITLE
Fix constant time bug in prp

### PR DESCRIPTION
**Pull request by coderdan**
This fixes an issue in the PRP `permute` function which resulted in non-constant time execution. This change ensures that permuting occurs in constant time.

### CODE REVIEW

File: src/primitives/prp.rs [hunk 1]
```diff
@@ -7,6 +7,18 @@ pub struct KnuthShufflePRP<T, const N: usize> {
...
 }
 
+fn const_position(input: &[u8], val: u8) -> Option<usize> {
+    let mut position = None;
+
+    for (index, elem) in input.iter().enumerate() {
+        if *elem == val && position == None {
+            position = Some(index);
+        }
+    }
```
Comment (freshtonic) [inline src/primitives/prp.rs:17]: How can we tell the compiler that we don't want that loop to be short circuited, ever? A smart compiler might undo our intent.

**Review by freshtonic — APPROVED** [review]
Approved with question.

File: src/primitives/prp.rs (hunk 1 above)
Comment (coderdan) [reply src/primitives/prp.rs:17]: Yeah I reckon LLVM is going to fight this pretty hard. We might need to look at doing it in assembly.

Comment (coderdan) [reply src/primitives/prp.rs:17]: There is an example approach here: https://github.com/isislovecruft/subtle/blob/master/src/lib.rs

**Review by freshtonic — APPROVED** [review]
Nice - correct and faster - pick 2!

**Pull request merged**